models/stdp_synapse.h

```diff
@@ -217,7 +217,7 @@ inline void
 stdp_synapse< targetidentifierT >::send( Event& e, thread t, const CommonSynapseProperties& )
 {
   // synapse STDP depressing/facilitation dynamics
-  const double t_spike = e.get_stamp().get_ms();
+  const double t_spike = e.get_stamp().get_ms() - e.get_offset();
```

Comment:
@clinssen This change makes me wonder if we have the same error in other synapse models as well.

Reply:
Yes, it does:
https://github.com/nest/nest-simulator/blob/720f4c55fec83b2925792dcd936e2d18eb076938/models/clopath_synapse.h#L215
https://github.com/nest/nest-simulator/blob/720f4c55fec83b2925792dcd936e2d18eb076938/models/ht_synapse.h#L189
https://github.com/nest/nest-simulator/blob/720f4c55fec83b2925792dcd936e2d18eb076938/models/jonke_synapse.h#L300
https://github.com/nest/nest-simulator/blob/720f4c55fec83b2925792dcd936e2d18eb076938/models/quantal_stp_synapse.h#L195
https://github.com/nest/nest-simulator/blob/720f4c55fec83b2925792dcd936e2d18eb076938/models/stdp_dopamine_synapse.h#L529
https://github.com/nest/nest-simulator/blob/720f4c55fec83b2925792dcd936e2d18eb076938/models/stdp_nn_pre_centered_synapse.h#L235
https://github.com/nest/nest-simulator/blob/720f4c55fec83b2925792dcd936e2d18eb076938/models/stdp_nn_restr_synapse.h#L229
https://github.com/nest/nest-simulator/blob/720f4c55fec83b2925792dcd936e2d18eb076938/models/stdp_nn_symm_synapse.h#L231
https://github.com/nest/nest-simulator/blob/720f4c55fec83b2925792dcd936e2d18eb076938/models/stdp_pl_synapse_hom.h#L244
https://github.com/nest/nest-simulator/blob/720f4c55fec83b2925792dcd936e2d18eb076938/models/stdp_synapse_hom.h#L281
https://github.com/nest/nest-simulator/blob/720f4c55fec83b2925792dcd936e2d18eb076938/models/stdp_triplet_synapse.h#L243
https://github.com/nest/nest-simulator/blob/720f4c55fec83b2925792dcd936e2d18eb076938/models/tsodyks2_synapse.h#L209
https://github.com/nest/nest-simulator/blob/720f4c55fec83b2925792dcd936e2d18eb076938/models/tsodyks_synapse_hom.h#L256
https://github.com/nest/nest-simulator/blob/720f4c55fec83b2925792dcd936e2d18eb076938/models/urbanczik_synapse.h#L207
https://github.com/nest/nest-simulator/blob/720f4c55fec83b2925792dcd936e2d18eb076938/models/vogels_sprekeler_synapse.h#L198

Reply:
@clinssen So this means that until now, we have not taken spike time offsets into account in plasticity mechanisms, doesn't it? It does not have any effect on the actual timing of the spike transmitted, since the offset information is passed on to the receiving neuron in the Event object. Could you create a follow-up issue for this? We should involve @abigailm in the discussion.